torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -673,6 +673,15 @@ std::unordered_map<std::string, std::string> ProcessGroupAgent::getMetrics() {
   return metrics;
 }
 
+TypeResolver ProcessGroupAgent::getTypeResolver() {
```

Comment:
Why is this part of the RpcAgent interface? It seems like this can just be a static utility function in rpc/utils.cpp?

Reply:
synced up offline, this is because we want to avoid introducing python dependency for files who call the getTypeResolver(), will create two new files: one file will have virtual getTypeResolver() interface that can be called by callers without python dependency, another file will have getTypeResolver() implementation that could have python dependency